### PR TITLE
Phase 2

### PR DESCRIPTION
**Pull request by manangouhari**
### Before you submit, please check that:

**What does this PR do?**
Integrate the content up until module-3 with the existing platform.

**Description of Task to be completed?**
Integrate the content up until module-3 with the existing platform.
The task included: 

- Implementation of curriculum page
- Dynamic generation of overview page for each individual module.
- Fixing the user flow to accomodate all the modules.

**How should this be manually tested?**
Go through the flow from start to finish.
Don't write the code, just keep on clicking next.

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -15,15 +16,28 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
...
-    actions.createPage({
-      path: `/lesson/${chapter.frontmatter.slug}`,
-      component: require.resolve('./src/templates/chapter.js'),
-      context: {
-        slug: chapter.frontmatter.slug,
-      },
-    });
+    if(file.frontmatter.type === "module"){
+      actions.createPage({
+        path: `/overview/${file.frontmatter.slug}`,
```
Comment (bhaskarSingh) [inline gatsby-node.js:25]: Add `tezos` to the prefix for the path for everything related to tezos.

File: src/templates/chapter-new.js [hunk 2]
```diff
@@ -0,0 +1,500 @@
+import React, { useState, useEffect } from 'react';
```
Comment (bhaskarSingh) [inline src/templates/chapter-new.js:1]: Where is the chapter-new template being used?

**Comment by bhaskarSingh** [comment]
Next steps:

1. Create an index page for showing all the modules.
2. Add compiler to the learning interface for module 2 & 3. Keep frontend validation logic for module-1 the same. Question to ask: How to go about implementing validation logic for every chapter within modules 2 & 3? Mock for reference: https://www.figma.com/file/sukJaz6yhGEVcqHF9HNZLz/BUIDL-Labs---Project---Cryptoverse-Wars-Abhinav-Pandey?node-id=0%3A1
3. Writing content for overview pages for modules 2 & 3.

@manangouhari  Let me know if I am missing something.

File: src/templates/chapter-new.js (hunk 2 above)
Comment (manangouhari) [reply src/templates/chapter-new.js:1]: Not being used right now. Was trying it out for experimentation earlier.

**Comment by manangouhari** [comment]
> Next steps:
> 
>     1. Create an index page for showing all the modules + linking everything from start to finish.
> 
>     2. Add compiler to the learning interface for module 2 & 3. Keep frontend validation logic for module-1 the same. Question to ask: How to go about implementing validation logic for every chapter within modules 2 & 3? Mock for reference: https://www.figma.com/file/sukJaz6yhGEVcqHF9HNZLz/BUIDL-Labs---Project---Cryptoverse-Wars-Abhinav-Pandey?node-id=0%3A1
> 
>     3. Writing content for overview pages for modules 2 & 3.
> 
> 
> @manangouhari Let me know if I am missing something.

Do we have a design for the index page?

**Comment by bhaskarSingh** [comment]
> Do we have a design for the index page?

No. You can refer cryptozombies [curriculum  ](https://cryptozombies.io/en/course/) design for inspiration.

File: gatsby-node.js [hunk 3]
```diff
@@ -15,15 +16,28 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
...
+        component: require.resolve('./src/templates/overview.js'),
+        context: {
+          slug: file.frontmatter.slug,
+        },
+        
+      });
+    }else{
+
+      actions.createPage({
+        path: `/lesson/${file.frontmatter.slug}`,
```
Comment (bhaskarSingh) [inline gatsby-node.js:35]: Missing `tezos` prefix

Comment (manangouhari) [reply gatsby-node.js:35]: Need to add it even for lessons?

Comment (bhaskarSingh) [reply gatsby-node.js:35]: Yup. Would keep the URLs in sync.

**Comment by bhaskarSingh** [comment]
TODO:

- Add content for a general overview of the lesson.
- Dynamically updating the header title of the learning interface according to the module.

**Comment by manangouhari** [comment]
> TODO:
> 
>     * Add content for a general overview of the lesson.
> 
>     * Dynamically updating the header title of the learning interface according to the module.

Have done the needful. Please review and let me know if there's something I missed.

**Pull request merged**